.github/workflows/checks.yml

```diff
@@ -54,14 +56,32 @@ jobs:
           echo git ${{ env.GIT_VERSION }}
           echo npm ${{ steps.npm-version.outputs.current-version }}
 
-      - name: Check if tag already exists
-        if: env.GIT_VERSION == steps.npm-version.outputs.current-version
+      - name: Get release commit messages
+        id: release-commit
         run: |
-          echo git tag for npm versions exists - please run `yarn version --patch` and update PR
-          exit 1
+          RELEASE_COMMIT_MESSAGE=$(git log --grep="Release-As: " -1 --pretty=format:"%s")
+          echo "RELEASE_COMMIT_MESSAGE=$RELEASE_COMMIT_MESSAGE" >> $GITHUB_ENV
 
-      - name: Tag available
-        run: echo versions not equal - running checks
+      - name: Check for existing tag and verify release commit
+        run: |
+          if [[ "${{ env.GIT_VERSION }}" == "${{ steps.npm-version.outputs.current-version }}" ]]; then
+            echo "Versions match. Proceeding to the next step."
+          else
+            echo "Versions do not match. Checking if tag exists."
+            tag_exists=$(git tag -l "v${{ steps.npm-version.outputs.current-version }}")
+            if [[ -n "$tag_exists" ]]; then
+              echo "Error: Tag 'v${{ steps.npm-version.outputs.current-version }}' already exists."
+              exit 1
+            else
+              echo "Tag does not exist. Verifying the release commit message."
+              if [[ "${{ env.RELEASE_COMMIT_MESSAGE }}" != *"chore: release ${{ steps.npm-version.outputs.current-version }}"* ]]; then
+                echo "Error: Expected commit message not found."
+                echo "Please run the following command to create the release commit:"
+                echo 'git commit --allow-empty -m "chore: release ${{ steps.npm-version.outputs.current-version }}" -m "Release-As: ${{ steps.npm-version.outputs.current-version }}"'
+                exit 1
+              fi
+            fi        
+          fi
```

Comment:
@shahmargi12 thanks for documentation! could you please still remove the empty commit part and update the documentation accordingly?
I think it's enough if we just document in the Release Process that the option to override the version proposed Release Please generally exists but the empty commit should not be part of the usual process.

I think we should remove the following steps in the pre-checks due to change in the release process as the plan is that from this change onwards we wouldn't necessarily release a new version with every merge into main anymore:
- Get latest git tag
- Remove tag 'v' prefix
- Get npm version
- Output versions
- Get release commit messages
- Check for existing tag and verify release commit

Reply:
ok sure!

and @evegufy , i also suggest to add build:storybook command in checks.yml. So that, if there are any issues with the Storybook build, they can be caught and fixed directly within the PR

Reply:
> @shahmargi12 thanks for the documentation! could you please still remove the empty commit part and update the documentation accordingly? I think it's enough if we just document in the Release Process that the option to override the version proposed Release Please generally exists but the empty commit should not be part of the usual process.
> 
> I think we should remove the following steps in the pre-checks due to change in the release process as the plan is that from this change onwards we wouldn't necessarily release a new version with every merge into main anymore:
> 
> * Get latest git tag
> * Remove tag 'v' prefix
> * Get npm version
> * Output versions
> * Get release commit messages
> * Check for existing tag and verify release commit

Hi @evegufy 
Above mentioned changes are done from my side. can you please review it ?

Reply:
> ok sure!
> 
> @evegufy , i also suggest to add build:storybook command in checks.yml. So that, if there are any issues with the Storybook build, they can be caught and fixed directly within the PR

sure, great idea!